supplier/spring-jdbc-supplier/src/main/java/org/springframework/cloud/fn/supplier/jdbc/JdbcSupplierConfiguration.java

```diff
@@ -47,45 +46,26 @@
 @EnableConfigurationProperties(JdbcSupplierProperties.class)
 public class JdbcSupplierConfiguration {
 
-	private final JdbcSupplierProperties properties;
-
-	private final DataSource dataSource;
-
-	public JdbcSupplierConfiguration(JdbcSupplierProperties properties, DataSource dataSource) {
-		this.properties = properties;
-		this.dataSource = dataSource;
-	}
-
 	@Bean
-	public MessageSource<Object> jdbcMessageSource(
+	public JdbcPollingChannelAdapter jdbcMessageSource(JdbcSupplierProperties properties, DataSource dataSource,
 			@Nullable ComponentCustomizer<JdbcPollingChannelAdapter> jdbcPollingChannelAdapterCustomizer) {
 
-		JdbcPollingChannelAdapter jdbcPollingChannelAdapter = new JdbcPollingChannelAdapter(this.dataSource,
-				this.properties.getQuery());
-		jdbcPollingChannelAdapter.setMaxRows(this.properties.getMaxRows());
-		jdbcPollingChannelAdapter.setUpdateSql(this.properties.getUpdate());
+		JdbcPollingChannelAdapter jdbcPollingChannelAdapter = new JdbcPollingChannelAdapter(dataSource,
+				properties.getQuery());
+		jdbcPollingChannelAdapter.setMaxRows(properties.getMaxRows());
+		jdbcPollingChannelAdapter.setUpdateSql(properties.getUpdate());
 		if (jdbcPollingChannelAdapterCustomizer != null) {
 			jdbcPollingChannelAdapterCustomizer.customize(jdbcPollingChannelAdapter);
 		}
 		return jdbcPollingChannelAdapter;
 	}
 
 	@Bean(name = "jdbcSupplier")
-	@PollableBean
```

Comment:
Nice catch on this removal being the only thing that was needing [this](https://github.com/spring-cloud/spring-functions-catalog/pull/128/files#diff-f096ec589da5515ddfb922db0df6b60552f7477d1a897d449b84d5312a1d987aL33) and hence allow the SC dep removal!